src/v/archival/ntp_archiver_service.cc

```diff
@@ -1478,12 +1478,37 @@ ntp_archiver::schedule_single_upload(const upload_context& upload_ctx) {
       upload.sources.empty()
       && upload_ctx.upload_kind == segment_upload_kind::compacted
       && model::offset{} != upload.final_offset) {
-        vlog(
-          _rtclog.warn,
-          "Upload skipped for range: {}-{} because these offsets lie inside "
-          "batches or the reupload is not smaller than the current segment",
+        const auto log_msg = fmt::format(
+          "Upload skipped for range: {}-{}, reason: {}",
           upload.starting_offset,
-          upload.final_offset);
+          upload.final_offset,
+          upload.status);
+
+        switch (upload.status) {
+        case upload_candidate::creation_status::success:
+            // we should never end up here
```

Comment:
nit: worth an assert?

Reply:
added, I thought there might be cases resulting in a successful candidate and the source list being empty, but now I see we return an error+empty candidate when there are no segments.